### PR TITLE
Closes #2332: Add WebAppManifestParser for creating WebAppManifest from JSON.

### LINKED ISSUE #2332: Parse JSON to create WebAppManifest
GeckoView will give us a `JSONObject` (that is already validated and cleaned up though!) - so we will need to create the `WebAppManifest` object from it.

### PR DESCRIPTION
**Pull request by pocmo**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/browser/session/src/main/java/mozilla/components/browser/session/manifest/WebAppManifestParser.kt [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        "standalone" -> WebAppManifest.DisplayMode.STANDALONE
+        "fullscreen" -> WebAppManifest.DisplayMode.FULLSCREEN
+        "minimal-ui" -> WebAppManifest.DisplayMode.MINIMAL_UI
+        "browser" -> WebAppManifest.DisplayMode.BROWSER
+        else -> WebAppManifest.DisplayMode.BROWSER
+    }
+}
+
+@Suppress("MagicNumber")
+private fun parseColor(color: String?): Int? {
```
Comment (Amejia481) [inline components/browser/session/src/main/java/mozilla/components/browser/session/manifest/WebAppManifestParser.kt]: nit: You could simplify the function a bit by combining the pre conditions.
```kotlin
    private fun parseColor(color: String?): Int? {
        if (color == null || !color.startsWith("#")) {
            return null
        }
        
        return try {
            Integer.parseInt(color.substring(1), 16)
        } catch (e: NumberFormatException) {
            null
        }
    }
```

File: components/browser/session/src/test/resources/manifests/spec_typical.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  "icons": [{
+    "src": "icon/lowres.webp",
+    "sizes": "64x64",
+    "type": "image/webp"
+  },{
+    "src": "icon/lowres.png",
+    "sizes": "64x64"
+  }, {
+    "src": "icon/hd_hi",
+    "sizes": "128x128"
```
Comment (Amejia481) [inline components/browser/session/src/test/resources/manifests/spec_typical.json]: Just for the sake of having more use cases tested, could we add a sample with an icon with multiple sizes? None of the manifest samples has it. Something like  ` "sizes": "128x128 192x192"`

**Comment by pocmo** [comment]
Okay, updated PR. :)
